lib/capistrano/tasks/migrations.rake

```diff
@@ -0,0 +1,123 @@
+namespace :forkcms do
+  namespace :migrations do
+    desc 'Prepare the server for running Fork CMS migrations'
+    task :prepare do
+      on roles(:web) do
+        create_migrations_file
+      end
+    end
+
+    desc 'Execute pending migrations'
+    task :execute do
+      on roles(:web) do
+        # Prepare for migrations if it's not done already
+        Rake::Task["forkcms:migrations:prepare"].invoke()
+
+        # If the current symlink doesn't exist yet, execute the first migration
+        execute_initial_migration if not test "[[ -e #{current_path} ]]"
+
+        # Abort if no migrations are found
+        migration_folders = get_migration_folders
+        if migration_folders.length == 0
+          next
+        end
+
+        # Fetch unprocessed migrations
+        migrations_to_execute = get_migrations_to_execute
+
+        # Abort if no new migrations are found
+        if migrations_to_execute.length == 0
+          next
+        end
+
+        # This can take a while and can go wrong. let's show a maintenance page
+        Rake::Task["forkcms:maintenance:enable"].invoke()
+
+        # Back up the database, just in case
+        Rake::Task["forkcms:database:backup"].invoke()
+
+        # Execute all migrations
+        migrations_to_execute.each do |dirname|
+          migration_path = "#{release_path}/migrations/#{dirname}"
+          migration_files = capture("ls -1 #{migration_path}").split(/\r?\n/)
+
+          migration_files.each do |filename|
+            # Update the locale through the console command
+            if filename.index('locale.xml') != nil
+              execute :php, "#{release_path}/bin/console forkcms:locale:import -f #{migration_path}/#{filename}"
+
+              next
+            end
+
+            # Execute a MySQL file
+            if filename.index('update.sql') != nil
+              Rake::Task["forkcms:database:execute"].invoke("#{migration_path}/#{filename}")
+
+              next
+            end
+          end
+        end
+
+        # All migrations were executed successfully and we didn't roll back so put them in the executed_migrations file
+        migrations_to_execute.each do |dirname|
+          execute :echo , "#{dirname} | tee -a #{shared_path}/executed_migrations"
```

Comment:
Hmm, wat als migration 1, 2 en 3 uitgevoerd zijn, maar 4 failed? Dan ga je de volgende keer 1, 2, en 3 terug uitvoeren?

Reply:
Pas wanneer _alle_ migrations succesvol zijn uitgevoerd worden deze weggeschreven naar `executed_migrations`. 
Wanneer er een migration faalt is het de bedoeling dat er een rollback gedaan wordt waardoor eventuele voorgaande migrations ongedaan gemaakt worden.

Dus ja, 1, 2 en 3 gaan opnieuw uitgevoerd worden maar wel op een 'verse' database

Reply:
Klopt, je hebt gelijk